File: src/main/java/gr/apt/controller/PersonRolesController.java
package gr.apt.controller;

import gr.apt.persistence.dto.PersonRolesDto;
import gr.apt.service.PersonRolesService;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import java.math.BigInteger;
import java.util.List;

@Path("/person-roles")
@Produces("application/json")
public class PersonRolesController {
    @Inject
    PersonRolesService service;

    @GET
    @Path("/list")
    public Response findAll(){
        List list = service.findAll();
        if(list != null){
            return Response.ok(list).build();
        }
        return Response.status(Response.Status.CONFLICT).build();
    }

    @GET
    @Path("/{id}")
    public Response findById(@PathParam("id") BigInteger id){
        PersonRolesDto dto = service.findById(id);
        if(dto != null){
            return Response.ok(dto).build();
        }
        return Response.status(Response.Status.CONFLICT).build();
    }

    @POST
    @Consumes("application/json")
    public Response create(PersonRolesDto dto){
        return service.create(dto) ? Response.ok(dto).build():Response.status(Response.Status.CONFLICT).build();
    }

    @PUT
    @Consumes("application/json")
    public Response update(PersonRolesDto dto){
        return service.update(dto) ? Response.ok(dto).build():Response.status(Response.Status.CONFLICT).build();
    }

    @DELETE
    @Consumes("application/json")
    public Response delete(PersonRolesDto dto){
        return service.delete(dto) ? Response.ok().build():Response.status(Response.Status.CONFLICT).build();
    }
}

File: src/main/java/gr/apt/controller/PersonRolesController.kt
package gr.apt.controller

import gr.apt.dto.PersonRolesDto
import gr.apt.service.PersonRolesService
import java.math.BigInteger
import javax.inject.Inject
import javax.ws.rs.*
import javax.ws.rs.core.Response

@Path("/person-roles")
@Produces("application/json")
class PersonRolesController {
    @get:Inject
    lateinit var service: PersonRolesService

    @GET
    @Path("/list")
    fun findAll(): Response {
        val list: List<*> = service.findAll()
        return Response.ok(list).build()
    }

    @GET
    @Path("/{id}")
    fun findById(@PathParam("id") id: BigInteger): Response {
        val dto = service.findById(id)
        return Response.ok(dto).build()
    }

    @POST
    @Consumes("application/json")
    fun create(dto: PersonRolesDto?): Response {
        return if (service.create(dto)) Response.ok(dto).build() else Response.status(Response.Status.CONFLICT)
            .build()
    }

    @PUT
    @Consumes("application/json")
    fun update(dto: PersonRolesDto): Response {
        return if (service.update(dto)) Response.ok(dto).build() else Response.status(Response.Status.CONFLICT)
            .build()
    }

    @DELETE
    @Consumes("application/json")
    fun delete(dto: PersonRolesDto): Response {
        return if (service.delete(dto)) Response.ok().build() else Response.status(Response.Status.CONFLICT).build()
    }
}